### DISCUSSION TITLE
How to send a React email in Express with Resend?

### DISCUSSION THREAD

**Discussion by masoncusack** [resend/react-email#1322] [Q&A] [answered]
Hi, in the [docs](https://react.email/docs/integrations/resend#3-send-email) the source code is provided for a file index.jsx which imports and sends an email, but it doesn't say how to run this. Can anyone provide a suggestion and maybe add this to the doc?

**Comment by gabrielmfern** [comment]
Its implicit, your "./email" is going to contain the React component you can use to send the email.
You'll just have to set up react and the types for it along with the appropriate [tsconfig setting](https://www.typescriptlang.org/tsconfig#jsx)

**Reply by gabrielmfern** [reply]
You can also send without the `react` option if that is what you want https://resend.com/docs/send-with-express#2-send-email-using-html.

**Reply by masoncusack** [reply]
Ahhh, that's exactly what I'm aiming to do, as I'd like to have a server that sends an email on receiving an http request.

Can I not use templates which are built and styled in react in this case, or just need to convert to HTML somehow?

**Reply by gabrielmfern** [reply]
you can use the React template by passing it on through the `react` option, the Resend SDK is already going to handle rendering it into HTML for you

**Reply by masoncusack** [reply]
And is it possible to import the react component into the express server?

**Reply by gabrielmfern** [reply, marked as answer]
Yeah, that's the general intent with the email components. React components aren't necessarily tied to the browser or the DOM, so we do the rendering into pure HTML since the React components are basically like a JSON object in the end.

So yes, you can import the React component into your express server as long as you have the types and JSX configured properly. Be aware that it will need a compilation step to use the JSX as well though, `tsc` will do the trick as long as you configure it right, as I mentioned before, but if you are using just plain JavaScript you will need something else.

**Comment by masoncusack** [comment]
Awesome, I will give it a go, thank you very much.
